### PR TITLE
Svelte - Shallow clone mutated object and array when diffing

### PR DESCRIPTION
**Pull request by Kiho**
I think it's not ideal to share objects between components

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "suppressImplicitAnyIndexErrors": true,
     "removeComments": false,
     "sourceMap": true,
+    "lib": [ "es2015" ],
```
Comment (matheusml) [inline tsconfig.json]: Why is this necessary?

Comment (Kiho) [reply tsconfig.json]: Because of Object.assign({}, val), you can suggest something else.

Comment (malbernaz) [reply tsconfig.json]: @Kiho you can replace it with `{ ...val }`. The compiler won't complain.

Comment (Kiho) [reply tsconfig.json]: @malbernaz Thank You!

**Comment by matheusml** [comment]
This is great @Kiho 

Could you please bump the npm version to 4.5.1 and add this change to the changelog file?

**Comment by matheusml** [comment]
Thanks @Kiho 💯
